vector-clock/vclock.c

```diff
@@ -0,0 +1,84 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "../master/slavelist.h"
+#include "vclock.h"
+
+/**
+ * Create a new vector clock.
+ *
+ * All times are initialized to `0`.
+ */
+vector_clock create_vclock(void)
+{
+    vector_clock return_val = (vector_clock)
+        malloc(sizeof(lamport_timestamp) * (NUM_MACHINES));
+
+    int i;
+    for (i = 0; i < NUM_MACHINES; i++) {
+        return_val[i] = 0;
+    }
+
+    return return_val;
+}
+
+/**
+ * Destroy the given vector clock.
+ */
+void destroy_vclock(vector_clock garbage)
+{
+    free(garbage);
+}
+
+/**
+ * Compare two vector clocks.
+ *
+ * Returns `-1` if `first` is earlier and `1` if `second` is earlier.
+ */
+int compare_vclock(vector_clock first, vector_clock second)
```

Comment:
if they are equal, shouldn't it return `0`?

Reply:
I can see you're going off the stackoverflow link I sent you, but do you think it should behave this way?

Reply:
Changed, that is how I was initially thinking of doing it.